src/main/java/me/jellysquid/mods/sodium/client/render/chunk/compile/ChunkBuilder.java

```diff
@@ -342,8 +342,18 @@ public void run() {
                     continue;
                 }
 
-                // Perform the build task with this worker's local resources and obtain the result
-                ChunkBuildResult result = job.task.performBuild(this.pipeline, this.bufferCache, job);
+                ChunkBuildResult result;
+                try {
+                    // Perform the build task with this worker's local resources and obtain the result
+                    result = job.task.performBuild(this.pipeline, this.bufferCache, job);
+                } catch (Exception e) {
+                    // Release resources attached to the failed task
```

Comment:
Perhaps it would be good to move `releaseResources` to a `finally` block, in order to reduce code duplication?